libs/blocks/ost/ost.js

```diff
@@ -38,46 +44,70 @@ export const createLinkMarkup = (defaults) => (
 ) => {
   const isCta = !!type?.startsWith('checkout');
 
-  const createHref = () => {
-    const params = new URLSearchParams([
-      ['osi', offerSelectorId],
-      ['type', type],
-    ]);
-    if (promo) params.set('promo', promo);
-    if (offer.commitment === 'PERPETUAL') params.set('perp', true);
-
-    if (isCta) {
-      const { workflow, workflowStep } = options;
-      params.set('text', options.ctaText ?? DEFAULT_CTA_TEXT);
-      if (workflow && workflow !== defaults.checkoutWorkflow) {
-        params.set('workflow', workflow);
-      }
-      if (workflowStep && workflowStep !== defaults.checkoutWorkflowStep) {
-        params.set('workflowStep', workflowStep);
-      }
-    } else {
-      const { displayRecurrence, displayPerUnit, displayTax, forceTaxExclusive } = options;
-      params.set('term', displayRecurrence);
-      params.set('seat', displayPerUnit);
-      params.set('tax', displayTax);
-      params.set('exclusive', forceTaxExclusive);
+  const params = new URLSearchParams([
+    ['osi', offerSelectorId],
+    ['type', type],
+  ]);
+  if (promo) params.set('promo', promo);
+  if (offer.commitment === 'PERPETUAL') params.set('perp', true);
+
+  if (isCta) {
+    const { workflow, workflowStep } = options;
+    params.set('text', options.ctaText ?? DEFAULT_CTA_TEXT);
+    if (workflow && workflow !== defaults.checkoutWorkflow) {
+      params.set('workflow', workflow);
     }
-    const { location } = window;
-    return `${location.protocol + location.host}/tools/ost?${params.toString()}`;
-  };
+    if (workflowStep && workflowStep !== defaults.checkoutWorkflowStep) {
+      params.set('workflowStep', workflowStep);
+    }
+  } else {
+    const {
+      displayRecurrence,
+      displayPerUnit,
+      displayTax,
+      forceTaxExclusive,
+    } = options;
+    params.set('term', displayRecurrence);
+    params.set('seat', displayPerUnit);
+    params.set('tax', displayTax);
+    params.set('exclusive', forceTaxExclusive);
+  }
+  const href = `https://milo.adobe.com/tools/ost?${params.toString()}`;
 
   const link = document.createElement('a');
   link.textContent = isCta
     ? `CTA {{${options.ctaText ?? DEFAULT_CTA_TEXT}}}`
     : `PRICE - ${offer.planType} - ${offer.name}`;
-  link.href = createHref();
+  link.href = href;
   return link;
 };
 
-export async function loadOstEnv() {
-  /* c8 ignore next */
-  const { Log, Defaults, getLocaleSettings } = await import('../../deps/commerce.js');
+const onSelectHandler = (createLinkMarkup) => (
+  offerSelectorId,
+  type,
+  offer,
+  options,
+  promoOverride,
+) => {
+  const link = createLinkMarkup(
+    offerSelectorId,
+    type,
+    offer,
+    options,
+    promoOverride,
+  );
+  if (!link) {
+    console.log('no link created, probably test context');
```

Comment:
We should consider disabling no-console for the whole file:
`/* eslint-disable no-console */`